src/uart/mod.rs

```diff
@@ -31,16 +32,58 @@ pub struct UART<Port = NoPort> {
 
 #[allow(unused)]
 impl UART<NoPort> {
-    pub fn port_0_init() -> UART<UART0> {
-        UART::<UART0>::init()
+    pub fn port_0_init(
+        baud_rate: BaudRates,
+        character_length: CharacterLength,
+        stop_bits: StopBits,
+        transmit_parity: bool,
+        parity_value: ParityValueSelect,
+        hfc: bool,
+    ) -> UART<UART0> {
+        UART::<UART0>::init(
+            baud_rate,
+            character_length,
+            stop_bits,
+            transmit_parity,
+            parity_value,
+            hfc,
+        )
     }
 
-    pub fn port_1_init() -> UART<UART1> {
-        UART::<UART1>::init()
+    pub fn port_1_init(
+        baud_rate: BaudRates,
+        character_length: CharacterLength,
+        stop_bits: StopBits,
+        transmit_parity: bool,
+        parity_value: ParityValueSelect,
+        hfc: bool,
+    ) -> UART<UART1> {
+        UART::<UART1>::init(
+            baud_rate,
+            character_length,
+            stop_bits,
+            transmit_parity,
+            parity_value,
+            hfc,
+        )
     }
 
-    pub fn port_2_init() -> UART<UART2> {
-        UART::<UART2>::init()
+    pub fn port_2_init(
+        baud_rate: BaudRates,
+        character_length: CharacterLength,
+        stop_bits: StopBits,
+        transmit_parity: bool,
+        parity_value: ParityValueSelect,
+        hfc: bool,
+    ) -> UART<UART2> {
+        UART::<UART2>::init(
+            baud_rate,
+            character_length,
+            stop_bits,
+            transmit_parity,
+            parity_value,
+            hfc,
+        )
```

Comment:
We should document how to use these to setup a basic `UART`. For example, it might not be entirely clear to the user what `hfc` is and how it affects communication. Mostly, these can just be copy and pasted doc comments. 

For example:
```rust
/// # My Function
/// My function will do basic function tasks like
/// adding two numbers and returning the result. 
///
/// ## Example
/// ```
/// let my_usize = my_func(2, 10);
/// assert_eq!(my_func(10, 2), my_usize);
/// ```
pub fn my_func(a: usize, b: usize) -> usize {
     a + b
}